pynest/nest/tests/test_connect_arrays_mpi.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+#
+# test_connect_arrays_mpi.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess as sp
+import unittest
+import nest
+import numpy as np
+
+try:
+    from mpi4py import MPI
+    HAVE_MPI4PY = True
+except ImportError:
+    HAVE_MPI4PY = False
+
+HAVE_MPI = nest.ll_api.sli_func("statusdict/have_mpi ::")
+MULTIPLE_PROCESSES = nest.NumProcesses() > 1
+
+
+@unittest.skipIf(not HAVE_MPI, 'NEST was compiled without MPI')
+@unittest.skipIf(not HAVE_MPI4PY, 'mpi4py is not available')
+class ConnectArraysMPICase(unittest.TestCase):
+    """
+    This TestCase uses mpi4py to collect and assert results from all
+    processes, and is supposed to only be run with multiple processes.
+    If running with nosetests or pytest, this TestCase is ignored when
+    run with a single process, and called from TestConnectArraysMPI using
+    multiple processes.
+    """
+    non_unique = np.array([1, 1, 3, 5, 4, 5, 9, 7, 2, 8], dtype=np.uint64)
+    comm = MPI.COMM_WORLD
+
+    # With nosetests, only run these tests if using multiple processes
+    __test__ = MULTIPLE_PROCESSES
```

Comment:
one can have ``MULTIPLE_PROCESSES`` False but ``HAVE_MPI`` True, would there then be some cases where the test would be run with a single process? (e.g. using pytest: I'm not sure why only nosetests is mentioned...)

More generally, one cannot have ``MULTIPLE_PROCESSES`` if ``HAVE_MPI`` is False so I'm not sure why ``HAVE_MPI`` is required.

Reply:
I've added `pytest` to the comment and removed the `skipIf(HAVE_MPI)`.